city_scrapers/spiders/fortx_Fort_Worth_Public_Meetings.py

```diff
@@ -0,0 +1,118 @@
+import json
+from datetime import datetime
+
+import scrapy
+from city_scrapers_core.constants import CITY_COUNCIL
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+from dateutil.parser import parse as dateparse
+
+
+class FortxFortWorthPublicMeetingsSpider(CityScrapersSpider):
+    name = "fortx_Fort_Worth_Public_Meetings"
+    agency = "Fort Worth Public Meetings"
+    timezone = "America/Chicago"
+    custom_settings = {
+        "ROBOTSTXT_OBEY": False,
+    }
+
+    main_url = "https://www.fortworthtexas.gov/"
+
+    meetings_url = "https://www.fortworthtexas.gov/ocapi/calendars/getcalendaritems"
+
+    meetings_url_payload = {
+        "LanguageCode": "en-US",
+        "Ids": ["8efac0b6-9ea3-402e-b7d9-e9e71a2a34a0"],
+        "StartDate": "",
+        "EndDate": "",
+    }
+
+    meeting_detail_url = (
+        "https://www.fortworthtexas.gov/ocapi/get/contentinfo"
+        "?calendarId={calendarId}&contentId={contentId}&language=en-US"
+        "&currentDateTime={currentDateTime}&mainContentId={mainContentId}"
+    )
+
+    def start_requests(self):
+        """
+        The meeting items for this organization are
+        fetched from two of its API endpoints. The
+        main API endpoint allows fetching meeting
+        items for the entirety of one year. The
+        spider is set to fetch all meetings for the
+        current year.
+        """
+        current_date = datetime.now().date()
```

Comment:
I ran the file `test_output.csv` and see only meetings of 2024 were scraped, while there are meetings in 2025 in the website's calendar.
Please adjust the scraping timeframe relatively to the current date, maybe from 6 months ago until the next 12 months.

Reply:
So there's a bit of a caveat to the API endpoint that is used for getting the meeting data for this organization. The endpoint's parameters require the start and end dates to be within the same year. This means we cannot fetch meetings spanning months from different years. Currently, the spider is set up to fetch all meetings for the year of the current date. When the year changes, it will automatically fetch meetings for the new year.

Alternatively, I could make two separate requests—one for each year. Let me know if this approach sounds better.